chalice/pipeline.py

```diff
@@ -9,18 +9,36 @@
 def create_pipeline_template(config):
     # type: (Config) -> Dict[str, Any]
     pipeline = CreatePipelineTemplate()
-    return pipeline.create_template(config.app_name)
+    return pipeline.create_template(config.app_name,
+                                    config.lambda_python_version)
+
+
+class InvalidCodeBuildPythonVersion(Exception):
+    def __init__(self, version):
+        # type: (str) -> None
+        super(InvalidCodeBuildPythonVersion, self).__init__(
+            'CodeBuild does not yet support python version %s.' % version
+        )
 
 
 class CreatePipelineTemplate(object):
 
+    _CODEBUILD_IMAGE = {
+        'python2.7': 'python:2.7.12',
+    }
+
     _BASE_TEMPLATE = {
         "AWSTemplateFormatVersion": "2010-09-09",
         "Parameters": {
             "ApplicationName": {
                 "Default": "ChaliceApp",
                 "Type": "String",
                 "Description": "Enter the name of your application"
+            },
+            "CodeBuildImage": {
```

Comment:
If we're going to parameterize the actual code build image anyways, shouldn't we let users also specify custom build images as well?  In other words, maybe we require the entire image identifier instead of always assuming `aws/codebuild/<param>`.